### PR TITLE
Components management

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

This PR introduce a cleaner way to define components for the Radium-Engine distribution and configure Core, IO, PluginBase, Gui and Engine as components of the RAdium package.
This is done by : 
 - modifying the cmake configuration scripts of all the base components : Core, Engine, IO, PluginBase and Gui
 - modifying the global project installation procedure.
 - defining an unified way to add new components to the Radium distribution

* **What is the current behavior?** (You can also link to an open issue here)

Components are configured globally exhibits some robustness issues (mainly concerning components inter-dependencies)

* **What is the new behavior (if this is a feature change)?**

Component can be configured individually and requested individually in a `find_package` command.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

* **Other information**:

### CODE REVIEW

File: tests/ExampleApps/HeadlessLibAndApp/libHeadless/RadiumHeadless/CLIBaseApplication.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include "CLIBaseApplication.hpp"
+
+int CLIBaseApplication::init( int argc, const char* argv[] ) {
+    (void)argc;
+    (void)argv;
```
Comment (nmellado) [inline tests/ExampleApps/HeadlessLibAndApp/libHeadless/RadiumHeadless/CLIBaseApplication.cpp:5]:
```suggestion
int CLIBaseApplication::init( int /*argc*/, const char* /*argv*/[] ) {
```

File: tests/ExampleApps/HeadlessLibAndApp/libHeadless/RadiumHeadless/CLIBaseApplication.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    return 0;
+}
+
+int CLIBaseApplication::run( float timeStep ) {
+    (void)timeStep;
```
Comment (nmellado) [inline tests/ExampleApps/HeadlessLibAndApp/libHeadless/RadiumHeadless/CLIBaseApplication.cpp:10]:
```suggestion
int CLIBaseApplication::run( float /*timeStep*/ ) {
```

**Review by dlyr — COMMENTED** [review]
Some wording in the documentation. I still need to test the generated doc.

**Comment by MathiasPaulin** [comment]
Doc updated, tested with doxygen 1.9.3 which still exhibit a problem with the awesomeCSS formatting. Except this, all is fine for me.

**Review by nmellado — APPROVED** [review]

**Pull request merged**